src/plugin.ts

```diff
@@ -0,0 +1,73 @@
+// import {
+//     inject,
+//     shallowRef,
+//     watch,
+//     toRaw,
+//     Vue2,
+//     isVue3,
+//     App,
+//     Plugin,
+//   } from "vue-demi";
+//   import { TonConnectUI, Wallet } from "@tonconnect/ui";
+//   import { isClientSide } from "../utils/web";
+//   import { TonConnectUIProviderProps } from "../utils/UIProvider";
+//   import { tonConnectUIKey } from "../injection-keys";
+  
+//   export const TonConnectUIPlugin: Plugin = {
+//     install(app: App | typeof Vue2, options?: TonConnectUIProviderProps) {
+//       const tonConnectUI = shallowRef<TonConnectUI | null>(null);
+//       const wallet = shallowRef<Wallet | null>(null);
+//       let unsubscribe = () => {};
+  
+//       if (!app) {
+//         throw new Error("TonConnectUI plugin is not installed");
+//       }
+  
+//       // Initialize TonConnectUI if on client side
+//       if (isClientSide() && !tonConnectUI.value) {
+//         tonConnectUI.value = new TonConnectUI(options || {});
+//       }
+  
+//       const updateTonConnectUIOptions = (newOptions: TonConnectUIProviderProps) => {
+//         if (isClientSide() && tonConnectUI.value) {
+//           tonConnectUI.value.uiOptions = toRaw(newOptions);
+//         }
+//       };
+  
+//       if (isVue3) {
```

Comment:
I'm not sure about plugin implementation right now. Does it work for you, did you test it? Because, lib build fails for me because of that file.

PS. I have to research how to make it compatible with both Vue 2 and Vue 3. Probably, plugin approach is more "vue-way".

PPS. Also, in case plugin is implemented, we should remove `<TonConnectUIProvider>` and update out "hooks" to use plugin. Also, docs should be updated.

Reply:
So I fixed plugin implementation. It works for both Vue2.7 and Vue3. Still have some problems with Nuxt2/3. But I suppose we can fix them during further testing before public release.

Reply:
@dnischeta Nice job 👍, there is a small issue, the format of codes are not right, it always have a space before each line, can you make a quick fix, then we can merge this into current branch?

Reply:
Done

Reply:
Thank you very much!